server/webhook/webhook.go

```diff
@@ -165,6 +165,15 @@ func normalizeNamespacedProject(pathWithNamespace string) (namespace string, pro
 	return strings.Join(splits[:len(splits)-1], "/"), splits[len(splits)-1]
 }
 
+// normalizeNamespacedProjectByHomepage converts data from web hooks to format expected by our plugin.
+func normalizeNamespacedProjectByHomepage(homapge string) (namespace string, project string) {
+	splits := strings.Split(homapge, "/")
+	if len(splits) < 2 {
+		return "", ""
```

Comment:
Also, I'm not super familiar with the problem domain here, but returning empty strings here feels potentially brittle.

Would it make sense to have `normalizeNamespacedProjectByHomepage` return the full namespace path (so `fmt.Sprintf("%s/%s", namespace, project)`) and in the event this case happens, just return `homepage` without further modification?

Reply:
make sense though the current function we have to get the subscribed channels for a projects need these are separated. Check [here](https://github.com/mattermost/mattermost-plugin-gitlab/pull/344/files#diff-b226a1832b33cac6c73b8efebbb63beb20d6547875952b3d0ecc9e80a01b4d60R47)